examples/webui-rest/main.c

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2022 Cesanta Software Limited
+// All rights reserved
+//
+// REST basics example
+// It implements the following endpoints:
+//    /api/f1 - respond with a simple mock result
+//    /api/sum - respond with the result of adding two numbers
+//    any other URI serves static files from s_root_dir
+// Results are JSON strings
+
+
+#include "mongoose.h"
+
+static const char *s_http_addr = "http://localhost:8000";  // HTTP port
+static const char *s_root_dir = "web_root";
+
+static void fn(struct mg_connection *c, int ev, void *ev_data, void *fn_data) {
+  if (ev == MG_EV_HTTP_MSG) {
+    struct mg_http_message *hm = (struct mg_http_message *) ev_data;
+    if (mg_http_match_uri(hm, "/api/f1")) {
+      mg_http_reply(c, 200, "Content-Type: application/json\r\n",
+                    "{\"result\": %d}\n", 123);
+    } else if (mg_http_match_uri(hm, "/api/sum")) {
+      // Attempt to fetch a JSON array from the body, hm->body
+      struct mg_str json = hm->body;
+      double num1, num2;
+      if (mg_json_get_num(json, "$[0]", &num1) &&
+          mg_json_get_num(json, "$[1]", &num2)) {
+        // Success! create a JSON response
+        char *resp = mg_mprintf("{%Q:%g}", "result", num1 + num2);
```

Comment:
That's not needed. We can printf directly to the output buffer, no need to have an intermediate `resp`:

```c
mg_http_reply(c, 200, "...headers...",  "{%Q:%g}\n", "result", num1 + num2);
```